src/main/java/Duke/Parser.java

```diff
@@ -0,0 +1,98 @@
+/**
+ * This class deals with making sense of the user command
+ */
+package Duke;
+
+import Command.Command;
+import Command.ExitCommand;
+import Command.ListCommand;
+import Command.MarkCommand;
+import Command.UnmarkCommand;
+import Command.DeleteCommand;
+import Command.TodoCommand;
+import Command.EventCommand;
+import Command.DeadlineCommand;
+import Command.FindCommand;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+import static java.lang.Integer.parseInt;
+
+public class Parser {
+
+    /**
+     * Returns the specific command that will execute the specific task according to
+     * the user input
+     * @@author shaniceng-reused
+     * Reused from https://github.com/tinenhao/ip/blob/master/src/main/java/Duke/Parser.java
+     * with minor modifications
+     *
+     * @param command
+     * @return Command that program should execute according to
+     *         the user input
+     * @throws DukeException to show the error of an invalid input or command
+     *         given by the user input
+     */
+    public static Command parse(String command) throws DukeException {
+        String[] input = command.split(" ", 2);
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd-MMM-yyyy HH:mm");
+        switch (input[0]) {
```

Comment:
May be nice to give input[0] a variable name too

Reply:
Okay! Thanks for highlighting :)